dali/operators/reader/video_reader_op.h

```diff
@@ -83,75 +97,97 @@ class VideoReader : public DataReader<GPUBackend, SequenceWrapper> {
   void SetupSharedSampleParams(DeviceWorkspace &ws) override {
   }
 
-  void RunImpl(DeviceWorkspace &ws) override {
-    auto& tl_sequence_output = ws.Output<GPUBackend>(0);
-    TensorList<GPUBackend> *label_output = NULL;
-    TensorList<GPUBackend> *frame_num_output = NULL;
-    TensorList<GPUBackend> *timestamp_output = NULL;
-
+  void SetOutputType(TensorList<GPUBackend> &output) {
     if (dtype_ == DALI_FLOAT) {
-      tl_sequence_output.set_type(TypeInfo::Create<float>());
+      output.set_type(TypeInfo::Create<float>());
     } else {  // dtype_ == DALI_UINT8
-      tl_sequence_output.set_type(TypeInfo::Create<uint8>());
+      output.set_type(TypeInfo::Create<uint8>());
     }
+  }
 
+  virtual void SetOutputShape(TensorList<GPUBackend> &output, DeviceWorkspace &ws) {
+    TensorListShape<> output_shape(batch_size_, sequence_dim);
     for (int data_idx = 0; data_idx < batch_size_; ++data_idx) {
-      auto sequence_shape = GetSample(data_idx).sequence.shape();
-      tl_shape_.set_tensor_shape(data_idx, sequence_shape);
+      output_shape.set_tensor_shape(
+        data_idx, GetSample(data_idx).sequence.shape());
     }
+    output.Resize(output_shape);
+  }
 
-    tl_sequence_output.Resize(tl_shape_);
-    tl_sequence_output.SetLayout("FHWC");
+  void PrepareVideoOutput(TensorList<GPUBackend> &output, DeviceWorkspace &ws) {
+    SetOutputType(output);
+    SetOutputShape(output, ws);
+    output.SetLayout("FHWC");
+  }
 
+  void PrepareAdditionalOutputs(DeviceWorkspace &ws) {
     if (enable_label_output_) {
       int output_index = 1;
-      label_output = &ws.Output<GPUBackend>(output_index++);
-      label_output->set_type(TypeInfo::Create<int>());
-      label_output->Resize(label_shape_);
+      label_output_ = &ws.Output<GPUBackend>(output_index++);
+      label_output_->set_type(TypeInfo::Create<int>());
```

Comment:
```suggestion
      label_output_->set_type(TypeTable::GetTypeInfo(TypeTable::GetTypeID<int>()););
```
It's rather better to use type table than create a new type info. 

BTW, TypeTable should definitely have a method to obtain a TypeInfo from a static type (basically those two methods in one).

Reply:
Done, added this method as: `TypeTable::GetTypeInfoFromStatic<int>()`